src/swarm/neo/protocol/connect/ConnectProtocol.d

```diff
@@ -519,8 +519,13 @@ class ConnectProtocol: ISelectClient
 
     override public void finalize ( FinalizeStatus status )
     {
+        static Exception finalize_exception;
+        if ( finalize_exception is null )
+            finalize_exception = new Exception("ConnectProtocol: finalized",
```

Comment:
ouch

Reply:
Ouch?

Reply:
Oh, ouch before the fix ;)

Reply:
Referring the the previous state of affairs.